src/server/express/middlewares/passport/openId.ts

```diff
@@ -0,0 +1,144 @@
+import * as OpenId from 'openid-client'
+import { gql, request } from 'graphql-request'
+import { UserInfo } from '../../../../types/expressSession'
+import {
+  OPENID_ISSUER,
+  OPENID_CLIENT_ID,
+  OPENID_CLIENT_SECRET,
+  GQL_BACKEND_URL,
+} from '../../../environment'
+
+import log from '../../../log'
+import {
+  AddUserMutation,
+  AddUserMutationVariables,
+  FindUserQuery,
+  FindUserQueryVariables,
+} from '../../../../generated/graphql'
+import { sign } from '../../../jwt'
+
+/**
+ * find the 3rd party (openid-connect) user in this app's system
+ * http://www.passportjs.org/docs/configure/#verify-callback
+ *
+ * @param _ tokenset not used
+ * @param userinfo
+ * @param done callback
+ */
+const verifyCallback = async (
+  _: OpenId.TokenSet,
+  {
+    given_name: givenName,
+    family_name: familyName,
+    emailAddress,
+    email = emailAddress as string,
+  }: OpenId.UserinfoResponse,
+  done: (error: unknown, user?: UserInfo) => void
+): Promise<void> => {
+  try {
+    if (!email || !familyName || !givenName) {
+      throw new Error(
+        `invalid user info ${JSON.stringify({
+          emailAddress,
+          familyName,
+          givenName,
+        })}`
+      )
+    }
+
+    const user = { email, familyName, givenName }
+    log.debug({ user }, 'openid verify Callback')
+
+    const Authorization = `Bearer ${sign({ email, roles: [] })}`
+    log.debug({ Authorization }, 'generated authorization')
+
+    const { user: existingUser } = await request<
```

Comment:
**Question**

My understanding is this is running on the server, so why are you needing to send a GraphQL request rather than using Prisma?

Reply:
Here are the reasons... let's discuss... what do you think?

I want to:
- centralize db access as much as possible
- enforce the authorization through JWT always.
- Decouple the authority (login), from actual API (gql), so that later we could split them

Reply:
Oh, sorry, I forgot to respond to this. I can understand your rationale here, and I'm good with it.